yktrace/src/lib.rs

```diff
@@ -67,12 +67,13 @@ impl IRTrace {
 
     pub fn compile(&self) {
         let len = self.len();
-        let func_names = self
-            .blocks
-            .iter()
-            .map(|b| b.func_name().as_ptr())
-            .collect::<Vec<_>>();
-        let bbs = self.blocks.iter().map(|b| b.bb()).collect::<Vec<_>>();
+        let mut func_names = Vec::with_capacity(len);
+        let mut bbs = Vec::with_capacity(len);
+        for i in 0..len {
```

Comment:
Is this `for blk in &self.blocks` in disguise?

Reply:
Fixed in 7b4c878